chiron/multistate.py

```diff
@@ -0,0 +1,968 @@
+        minimized_state = minimize_energy(
+            sampler_state.x0,
+            thermodynamic_state.potential.compute_energy,
+            self.nbr_list,
+            maxiter=max_iterations,
+        )
+
+        # Update the sampler state
+        self._sampler_states[replica_id].x0 = minimized_state.params
+
+        # Compute and log final energy
+        final_energy = thermodynamic_state.get_reduced_potential(sampler_state)
+        log.debug(
+            f"Replica {replica_id + 1}/{self.n_replicas}: final energy {final_energy:8.3f}kT"
+        )
+
+    def minimize(
+        self,
+        tolerance=1.0 * unit.kilojoules_per_mole / unit.nanometers,
+        max_iterations: int = 1_000,
+    ) -> None:
+        """
+        Minimize all replicas in the sampler.
+
+        This method minimizes the positions of all replicas to the nearest local
+        minimum of the potential energy surface. The minimized positions are stored
+        at the end of the process.
+
+        Parameters
+        ----------
+        tolerance : unit.Quantity, optional
+            The energy tolerance for the minimization. Default is 1.0 kJ/mol/nm.
+        max_iterations : int, optional
+            The maximum number of iterations for the minimization process.
+            Default is 1000.
+
+        Raises
+        ------
+        RuntimeError
+            If the simulation has not been created before calling this method.
+        """
+
+        # Check that simulation has been created.
+        if self.n_replicas == 0:
+            raise RuntimeError(
+                "Cannot minimize replicas. The simulation must be created first."
+            )
+
+        log.debug("Minimizing all replicas...")
+
+        # Iterate over all replicas and minimize them
+        for replica_id in range(self.n_replicas):
+            self._minimize_replica(replica_id, tolerance, max_iterations)
+
+    def _equilibration_timings(self, timer, iteration: int, n_iterations: int):
+        iteration_time = timer.stop("Equilibration Iteration")
+        partial_total_time = timer.partial("Run Equilibration")
+        time_per_iteration = partial_total_time / iteration
+        estimated_time_remaining = time_per_iteration * (n_iterations - iteration)
+        estimated_total_time = time_per_iteration * n_iterations
+        estimated_finish_time = time.time() + estimated_time_remaining
+        # TODO: Transmit timing information
+
+        log.info(f"Iteration took {iteration_time:.3f}s.")
+        if estimated_time_remaining != float("inf"):
+            log.info(
+                "Estimated completion (of equilibration only) in {}, at {} (consuming total wall clock time {}).".format(
+                    str(datetime.timedelta(seconds=estimated_time_remaining)),
+                    time.ctime(estimated_finish_time),
+                    str(datetime.timedelta(seconds=estimated_total_time)),
+                )
+            )
+
+    def equilibrate(
```

Comment:
We can probably get rid of `equilibrate` and equilibration-related stuff entirely to simplify things while we iterate. It was only intended to be used in case there was some equilibration needed with a shorter timestep for stability. Practically, it's best to just run a production simulation and figure out what (if anything) to discard later.